llvm/lib/CodeGen/TailDuplicator.cpp

```diff
@@ -669,27 +669,6 @@ bool TailDuplicator::shouldTailDuplicate(bool IsSimple,
       return false;
   }
 
-  // Check if any of the successors of TailBB has a PHI node in which the
-  // value corresponding to TailBB uses a subregister.
-  // If a phi node uses a register paired with a subregister, the actual
-  // "value type" of the phi may differ from the type of the register without
-  // any subregisters. Due to a bug, tail duplication may add a new operand
-  // without a necessary subregister, producing an invalid code. This is
-  // demonstrated by test/CodeGen/Hexagon/tail-dup-subreg-abort.ll.
```

Comment:
Can you remove the REQUIRES: asserts in this test, and add some generated checks? We could also write a targeted MIR test to make sure this works now